webapp/app.py

```diff
@@ -0,0 +1,66 @@
+"""
+A Flask application for maas.io
+"""
+
+# Packages
+import flask
+import talisker.flask
+
+from canonicalwebteam.yaml_responses.flask_helpers import (
+    prepare_deleted,
+    prepare_redirects,
+)
+from werkzeug.contrib.fixers import ProxyFix
+from werkzeug.debug import DebuggedApplication
+
+from webapp.feeds import get_rss_feed
+from webapp.helpers import get_current_year, versioned_static
+from webapp.views import TemplateFinder
+
+
+app = flask.Flask(
+    __name__, template_folder="../templates", static_folder="../static"
+)
+
+app.url_map.strict_slashes = False
+
+app.wsgi_app = ProxyFix(app.wsgi_app)
+if app.debug:
+    app.wsgi_app = DebuggedApplication(app.wsgi_app)
+
+
+talisker.flask.register(app)
+talisker.logs.set_global_extra({"service": "maas.io"})
+
+
+app.before_request(prepare_redirects())
+app.before_request(prepare_deleted())
+
+
+template_finder_view = TemplateFinder.as_view("template_finder")
+app.add_url_rule("/", view_func=template_finder_view)
+app.add_url_rule("/<path:subpath>", view_func=template_finder_view)
+
+
+@app.errorhandler(404)
+def not_found_error(error):
+    return flask.render_template("404.html"), 404
+
+
+@app.errorhandler(500)
+def internal_error(error):
+    return flask.render_template("500.html"), 500
+
+
+@app.context_processor
+def context():
+    return {
+        "ASSET_SERVER_URL": "https://assets.ubuntu.com/v1/",
```

Comment:
I'm not sure about `ASSET_SERVER_URL` here. I like URLs to be complete in templates if possible, so they can be easily grepped for, or clicked on / pasted into a browser. There should be a reason for breaking a URL. The assets server URL hasn't changed since the asset server was created 5 years ago, and I don't see it changing soon. And even if it did change, a global find-and-replace on the project is not really very hard to do.

So I'd prefer we just do `find templates -name '*.html' -exec sed -i 's!{{ ASSET_SERVER_URL }}!https://assets.ubuntu.com/v1/!g' {} \;` and remove this global variable.

Thoughts?

Reply:
If we are confident it will not change, I don't mind just having it in the templates and removing the constant. I just added it because we were doing it with django, happy to remove if we think it's the way to go.